sounds/views.py

```diff
@@ -318,22 +318,13 @@ def sound_download(request, username, sound_id):
     if sound.user.username.lower() != username.lower():
         raise Http404
 
-    if settings.LOG_DOWNLOADS:
-        range_header = request.META.get('HTTP_RANGE', '').replace('bytes=', '').split('-')
-        if 'HTTP_RANGE' not in request.META or range_header[0] == "0":
-            downloads_logger.info('Download sound', extra={
-                'user_id': request.user.id,
-                'user_ip': request.META.get('HTTP_X_FORWARDED_FOR'),
-                'protocol': request.META.get('HTTP_X_FORWARDED_PROTOCOL'),
-                'session_id': request.session.session_key,
-                'user_agent': request.META.get('HTTP_USER_AGENT'),
-                'method': request.method,
-                'sound_id': sound_id,
-                'range': request.META.get('HTTP_RANGE', None),
-            })
-
-    if not Download.objects.filter(user=request.user, sound=sound).exists():
-        Download.objects.create(user=request.user, sound=sound, license=sound.license)
+    if 'HTTP_RANGE' not in request.META:
+        cache_key = 'sdwn_%s_%d' % (sound_id, request.user.id)
+        if cache.get(cache_key, None) == None:
+            if not Download.objects.filter(user=request.user, sound=sound).exists():
```

Comment:
This only creates a download object if it doesn't already exist. I thought that this patch was to add multiple downloads :confused:

Reply:
on top of that, with the number of downloads we have, won't this flood the memcache?

Reply:
You're right @alastair, this `if` should not be here.
And regarding @bdejong comment, we did not think about this when we discussed the best way to implement this feature. Hopefully not, it's not that much data to save and it's only for 5 minutes.